### PR TITLE
Introduce new CLI

### LINKED ISSUE #32: Use run_command_and_stop instead of run_simple
While running the tests we get the following warnings:

`The use of "#run_simple" is deprecated. Use "run_command_and_stop" instead. Called by ./spec/cobra_commander/cli_spec.rb:111:in `block (4 levels) in <top (required)>'`

We should update this in order to avoid future problems when updating gems.

### PR DESCRIPTION
**Pull request by xjunior**
## Refator CLI to open possibilities

With this new CLI actions there were previously directed only to umbrella are now available to components, such as `cobra graph`, `cobra tree`, `cobra ls`.

You're now also able to work on top of `js` or `ruby` dependencies only (`--js`, `--ruby`), and filter by `--dependencies` and `--dependents` of a component when appropriate, allowing constructions like:

`cobra exec --ruby --dependents core_models 'pwd'`
```bash
bundle exec cobra exec --app /Users/chjunior/workspace/power/nitro --ruby --dependents core_models pwd
===> user_profile (/Users/chjunior/workspace/power/nitro/components/user_profile)
/Users/chjunior/workspace/power/nitro/components/user_profile
===> allowances (/Users/chjunior/workspace/power/nitro/components/allowances)
/Users/chjunior/workspace/power/nitro/components/allowances
===> accounting (/Users/chjunior/workspace/power/nitro/components/accounting)
/Users/chjunior/workspace/power/nitro/components/accounting
===> product_rules (/Users/chjunior/workspace/power/nitro/components/product_rules)
/Users/chjunior/workspace/power/nitro/components/product_rules
===> mdm (/Users/chjunior/workspace/power/nitro/components/mdm)
/Users/chjunior/workspace/power/nitro/components/mdm
===> equipment_assets (/Users/chjunior/workspace/power/nitro/components/equipment_assets)
/Users/chjunior/workspace/power/nitro/components/equipment_assets
===> nitro_mobile_platform (/Users/chjunior/workspace/power/nitro/components/nitro_mobile_platform)
/Users/chjunior/workspace/power/nitro/components/nitro_mobile_platform
===> aerial_measurements (/Users/chjunior/workspace/power/nitro/components/aerial_measurements)
/Users/chjunior/workspace/power/nitro/components/aerial_measurements
===> nitro_notifications (/Users/chjunior/workspace/power/nitro/components/nitro_notifications)
/Users/chjunior/workspace/power/nitro/components/nitro_notifications
===> employee_referrals (/Users/chjunior/workspace/power/nitro/components/employee_referrals)
/Users/chjunior/workspace/power/nitro/components/employee_referrals
===> connect_and_corporate_events_models (/Users/chjunior/workspace/power/nitro/components/connect_and_corporate_events_models)
/Users/chjunior/workspace/power/nitro/components/connect_and_corporate_events_models
===> apm (/Users/chjunior/workspace/power/nitro/components/apm)
/Users/chjunior/workspace/power/nitro/components/apm
===> sales (/Users/chjunior/workspace/power/nitro/components/sales)
/Users/chjunior/workspace/power/nitro/components/sales
===> badge_access (/Users/chjunior/workspace/power/nitro/components/badge_access)
/Users/chjunior/workspace/power/nitro/components/badge_access
===> support_ticket_model (/Users/chjunior/workspace/power/nitro/components/support_ticket_model)
/Users/chjunior/workspace/power/nitro/components/support_ticket_model
===> nitro_survey (/Users/chjunior/workspace/power/nitro/components/nitro_survey)
/Users/chjunior/workspace/power/nitro/components/nitro_survey
===> business_intelligence (/Users/chjunior/workspace/power/nitro/components/business_intelligence)
/Users/chjunior/workspace/power/nitro/components/business_intelligence
===> nitro_graphql (/Users/chjunior/workspace/power/nitro/components/nitro_graphql)
/Users/chjunior/workspace/power/nitro/components/nitro_graphql
===> expenses (/Users/chjunior/workspace/power/nitro/components/expenses)
/Users/chjunior/workspace/power/nitro/components/expenses
===> mobile_devices (/Users/chjunior/workspace/power/nitro/components/mobile_devices)
/Users/chjunior/workspace/power/nitro/components/mobile_devices
===> training (/Users/chjunior/workspace/power/nitro/components/training)
/Users/chjunior/workspace/power/nitro/components/training
===> human_resources (/Users/chjunior/workspace/power/nitro/components/human_resources)
/Users/chjunior/workspace/power/nitro/components/human_resources
===> call_transcriptions (/Users/chjunior/workspace/power/nitro/components/call_transcriptions)
/Users/chjunior/workspace/power/nitro/components/call_transcriptions
===> spaces (/Users/chjunior/workspace/power/nitro/components/spaces)
/Users/chjunior/workspace/power/nitro/components/spaces
===> email_templates (/Users/chjunior/workspace/power/nitro/components/email_templates)
/Users/chjunior/workspace/power/nitro/components/email_templates
===> admin (/Users/chjunior/workspace/power/nitro/components/admin)
/Users/chjunior/workspace/power/nitro/components/admin
===> powerlife (/Users/chjunior/workspace/power/nitro/components/powerlife)
/Users/chjunior/workspace/power/nitro/components/powerlife
===> installation_scheduler (/Users/chjunior/workspace/power/nitro/components/installation_scheduler)
/Users/chjunior/workspace/power/nitro/components/installation_scheduler
===> announcements (/Users/chjunior/workspace/power/nitro/components/announcements)
/Users/chjunior/workspace/power/nitro/components/announcements
===> flight_control (/Users/chjunior/workspace/power/nitro/components/flight_control)
/Users/chjunior/workspace/power/nitro/components/flight_control
===> adp (/Users/chjunior/workspace/power/nitro/components/adp)
/Users/chjunior/workspace/power/nitro/components/adp
===> nitro_component_transition (/Users/chjunior/workspace/power/nitro/components/nitro_component_transition)
/Users/chjunior/workspace/power/nitro/components/nitro_component_transition
===> home_tour (/Users/chjunior/workspace/power/nitro/components/home_tour)
/Users/chjunior/workspace/power/nitro/components/home_tour
===> corporate_events (/Users/chjunior/workspace/power/nitro/components/corporate_events)
/Users/chjunior/workspace/power/nitro/components/corporate_events
===> compensation (/Users/chjunior/workspace/power/nitro/components/compensation)
/Users/chjunior/workspace/power/nitro/components/compensation
===> sso (/Users/chjunior/workspace/power/nitro/components/sso)
/Users/chjunior/workspace/power/nitro/components/sso
===> purchase_orders (/Users/chjunior/workspace/power/nitro/components/purchase_orders)
/Users/chjunior/workspace/power/nitro/components/purchase_orders
===> support (/Users/chjunior/workspace/power/nitro/components/support)
/Users/chjunior/workspace/power/nitro/components/support
===> link_share (/Users/chjunior/workspace/power/nitro/components/link_share)
/Users/chjunior/workspace/power/nitro/components/link_share
===> charitable_giving (/Users/chjunior/workspace/power/nitro/components/charitable_giving)
/Users/chjunior/workspace/power/nitro/components/charitable_giving
===> inventory (/Users/chjunior/workspace/power/nitro/components/inventory)
/Users/chjunior/workspace/power/nitro/components/inventory
===> contact_center (/Users/chjunior/workspace/power/nitro/components/contact_center)
/Users/chjunior/workspace/power/nitro/components/contact_center
===> marketing (/Users/chjunior/workspace/power/nitro/components/marketing)
/Users/chjunior/workspace/power/nitro/components/marketing
===> finance (/Users/chjunior/workspace/power/nitro/components/finance)
/Users/chjunior/workspace/power/nitro/components/finance
===> tournaments (/Users/chjunior/workspace/power/nitro/components/tournaments)
/Users/chjunior/workspace/power/nitro/components/tournaments
===> connect (/Users/chjunior/workspace/power/nitro/components/connect)
/Users/chjunior/workspace/power/nitro/components/connect
===> dev_docs (/Users/chjunior/workspace/power/nitro/components/dev_docs)
/Users/chjunior/workspace/power/nitro/components/dev_docs
===> call_queues (/Users/chjunior/workspace/power/nitro/components/call_queues)
/Users/chjunior/workspace/power/nitro/components/call_queues
===> engineering (/Users/chjunior/workspace/power/nitro/components/engineering)
/Users/chjunior/workspace/power/nitro/components/engineering
===> travel (/Users/chjunior/workspace/power/nitro/components/travel)
/Users/chjunior/workspace/power/nitro/components/travel
===> recruiting (/Users/chjunior/workspace/power/nitro/components/recruiting)
/Users/chjunior/workspace/power/nitro/components/recruiting
===> solar_roofing (/Users/chjunior/workspace/power/nitro/components/solar_roofing)
/Users/chjunior/workspace/power/nitro/components/solar_roofing
===> power_glossary (/Users/chjunior/workspace/power/nitro/components/power_glossary)
/Users/chjunior/workspace/power/nitro/components/power_glossary
===> attendance (/Users/chjunior/workspace/power/nitro/components/attendance)
/Users/chjunior/workspace/power/nitro/components/attendance
```

Which would execute `bundle` on each core models dependency, or even:

`cobra exec core_models 'bundle'`

```bash
➜  cobra_commander git:(refactor-cli) bundle exec cobra exec --app /Users/chjunior/workspace/power/nitro core_models bundle
===> core_models (/Users/chjunior/workspace/power/nitro/components/core_models)
Your Gemfile lists the gem nitro_linting (>= 0) more than once.
You should probably keep only one of them.
Remove any duplicate entries and specify the gem only once (per group).
While it's not a problem now, it could cause errors if you change the version of one of them later.
Using rake 11.3.0
Using Ascii85 1.0.3
Using aasm 3.4.0
Using concurrent-ruby 1.1.6
(…)
Bundle complete! 65 Gemfile dependencies, 379 gems now installed.
Bundled gems are installed into `/Users/chjunior/workspace/power/nitro/vendor/bundle`
```

To execute bundle in core_models.

Or even:

`cobra graph --ruby core_models`

To get a graph of all ruby dependencies of core models:

output

### cobra help

```sh
Commands:
  cobra changes [--results=RESULTS] [--branch=BRANCH]  # Prints list of changed files
  cobra exec [component] <command>                     # Executes the command in the context of a given component or set of components. If no component is given executes the command in all components.
  cobra graph [component]                              # Outputs a graph of a given component or umbrella
  cobra help [COMMAND]                                 # Describe available commands or one specific command
  cobra ls [component]                                 # Lists the components in the context of a given component or umbrella
  cobra tree [component]                               # Prints the dependency tree of a given component or umbrella
  cobra version                                        # Prints version

Options:
  -a, [--app=APP]            
                             # Default: /Users/chjunior/workspace/power/cobra_commander
      [--js], [--no-js]      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]  # Consider only the Ruby dependency graph

```

### cobra changes

```sh
➜  nitro git:(nova/remove-cobra-commander) be cobra help changes
Usage:
  cobra changes [--results=RESULTS] [--branch=BRANCH]

Options:
  -r, [--results=RESULTS]    # Accepts test, full, name or json
                             # Default: test
  -b, [--branch=BRANCH]      # Specified target to calculate against
                             # Default: master
  -a, [--app=APP]
                             # Default: /Users/chjunior/workspace/power/nitro
      [--js], [--no-js]      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]  # Consider only the Ruby dependency graph

Prints list of changed files
```

### cobra exec

```sh
Usage:
  cobra exec [component] <command>

Options:
      [--dependencies], [--no-dependencies]  # Run the command on each dependency of a given component
      [--dependents], [--no-dependents]      # Run the command on each dependency of a given component
  -a, [--app=APP]                            
                                             # Default: /Users/chjunior/workspace/power/cobra_commander
      [--js], [--no-js]                      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]                  # Consider only the Ruby dependency graph

Executes the command in the context of a given component or set of components. If no component is given executes the command in all components.
```

### cobra graph

```sh
Usage:
  cobra graph [component]

Options:
  -o, [--output=OUTPUT]      # Output file, accepts .png or .dot
                             # Default: /Users/chjunior/workspace/power/cobra_commander/output.png
  -a, [--app=APP]            
                             # Default: /Users/chjunior/workspace/power/cobra_commander
      [--js], [--no-js]      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]  # Consider only the Ruby dependency graph

Outputs a graph of a given component or umbrella
```

### cobra ls

```sh
Usage:
  cobra ls [component]

Options:
  -d, [--dependencies], [--no-dependencies]  # Run the command on each dependency of a given component
  -D, [--dependents], [--no-dependents]      # Run the command on each dependency of a given component
  -t, [--total], [--no-total]                # Prints the total count of components
  -a, [--app=APP]                            
                                             # Default: /Users/chjunior/workspace/power/cobra_commander
      [--js], [--no-js]                      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]                  # Consider only the Ruby dependency graph

Lists the components in the context of a given component or umbrella
```

### cobra tree

```sh
Usage:
  cobra tree [component]

Options:
  -a, [--app=APP]            
                             # Default: /Users/chjunior/workspace/power/cobra_commander
      [--js], [--no-js]      # Consider only the JS dependency graph
      [--ruby], [--no-ruby]  # Consider only the Ruby dependency graph

Prints the dependency tree of a given component or umbrella
```

Resolves #32

### CODE REVIEW

**Comment by drborges** [comment]
Beautiful 👏🏼

File: spec/fixtures/app/components/f/package.json [hunk 1]
```diff
@@ -3,5 +3,8 @@
   "version": "0.0.1",
   "description": "Some description",
   "author": "Garett Arrowood",
-  "license": "UNLICENSED"
+  "license": "UNLICENSED",
```
Comment (wadewinningham) [inline spec/fixtures/app/components/f/package.json:6]: This really *Unlicensed*? Not relevant to this PR but something we should address at some point.

**Comment by wadewinningham** [comment]
Does README.md need to be updated, as well?

Would love to see some example output for `tree`, `ls`, and `changes` in the PR description. Even if it's the spec output.

**Comment by wadewinningham** [comment]
Also, does any of this work resolve the [issues](https://github.com/powerhome/cobra_commander/issues) currently open?

**Comment by xjunior** [comment]
@wadewinningham the outputs didn't change from previous version. I'll post how they look for a single component. This PR resolves #32, I updated the description.

**Comment by xjunior** [comment]
@wadewinningham done

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## Version 0.8 - 2020-07-21
+
```
Comment (benlangfeld) [inline CHANGELOG.md:6]: Releases get cut on master.

```suggestion
```

File: lib/cobra_commander/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module CobraCommander
-  VERSION = "0.7.0"
+  VERSION = "0.8.0"
```
Comment (benlangfeld) [inline lib/cobra_commander/version.rb:4]: Don't bump the version here, bump it separately once this is on master.

**Comment by benlangfeld** [comment]
@xjunior https://travis-ci.org/github/powerhome/cobra_commander/builds/712214612

Not sure why the check isn't showing up for this.

**Pull request merged**